packages/appsync-modelgen-plugin/src/__tests__/visitors/__snapshots__/appsync-json-metadata-visitor.test.ts.snap

```diff
@@ -317,6 +317,7 @@ exports[`Metadata visitor for custom PK support relation metadata for hasMany &
     },
     \\"enums\\": {},
     \\"nonModels\\": {},
+    \\"pragma\\": \\"2.1.0\\",
```

Comment:
can you please tell me how this will be used?

Reply:
The new version of the library will not be compatible with the old version of codegen. This will be used for the library to determine what version of codegen was used. In the case of the new version of the library with the old version of codegen there will be an error along the lines of `Codegen version x.x.x is not supported please use ^x.x.x`.

Reply:
Naming isn't set in stone. `version` was already taken, but we could do something like `codegenVersion`.

Reply:
Some thoughts on the `pragma`:  Although the `appsync-modelgen-plugin` package version has direct relation with the library, as a CLI user it is not easy to directly consume this package. The current plugin relation is `amplify-cli => amplify-codegen => appsync-modelgen-plugin`.  My suggestion is to use the version of `amplify-codegen` for this value as it is within the same mono-repo structure and have fixed dependency of `appsync-modelgen-plugin`

Reply:
Alternatively we can figure out a way to determine the specific `amplify-codegen` version given the existing `appsync-modelgen-plugin` package

Reply:
Makes sense, I think it could be passed as config here then: https://github.com/aws-amplify/amplify-codegen/blob/main/packages/amplify-codegen/src/commands/models.js#L119.

Reply:
If you're using this to store codegen version to check against in the library, I would suggest re-naming this to `codegenVersion` to make it more obvious and also use the version of the `amplify-codegen` package like Aaron mentioned.